src/main/java/appeng/tile/crafting/TileMolecularAssembler.java

```diff
@@ -338,58 +342,31 @@ public void updateSetting( final IConfigManager manager, final Enum settingName,
 	}
 
 	@Override
-	public IInventory getInternalInventory()
+	public IItemHandlerModifiable getInternalInventory()
 	{
-		return this.inv;
+		return this.internalInv;
 	}
 
 	@Override
-	public int getInventoryStackLimit()
+	protected IItemHandler getItemHandlerForSide(EnumFacing side)
 	{
-		return 1;
-	}
-
-	@Override
-	public boolean isItemValidForSlot( final int i, final ItemStack itemstack )
-	{
-		if( i >= 9 )
-		{
-			return false;
-		}
-
-		if( this.hasPattern() )
-		{
-			return this.myPlan.isValidItemForSlot( i, itemstack, this.getWorld() );
-		}
-
-		return false;
+		return gridInv;
 	}
-
+	
 	private boolean hasPattern()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this method into "CraftingGridFilter". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3398)